src/coreclr/vm/jitinterface.cpp

```diff
@@ -11636,7 +11637,19 @@ InfoAccessType CEEJitInfo::constructStringLiteral(CORINFO_MODULE_HANDLE scopeHnd
     }
     else
     {
-        *ppValue = (LPVOID)ConstructStringLiteral(scopeHnd, metaTok); // throws
+        void** ptr = (void**)ConstructStringLiteral(scopeHnd, metaTok); // throws
+
+        FrozenObjectHeap* foh = SystemDomain::GetSegmentWithFrozenObjects();
```

Comment:
This should be returned from `ConstructStringLiteral` and not detected after the fact.

Reply:
I would make the `ConstructStringLiteral` signature to look like this:
```
// ppPinnedString: If the string is pinned (e.g. allocated in frozen heap), the pointer to the pinned string is returned in *ppPinnedPointer. ppPinnedPointer == nullptr
// means that the caller does not care whether the string is pinned or not.

ConstructStringLiteral(CORINFO_MODULE_HANDLE scopeHnd, mdToken metaTok, void** ppPinnedString = nullptr);
```